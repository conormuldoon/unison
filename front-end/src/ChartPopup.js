import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { IoMdClose } from "react-icons/io";
import './App.css';
import { PRECIP } from './Constant';
import TabsComponent from './TabsComponent';
import { expandLink, chartText } from './Util';


const DLEN = 16;

/**
 * A popup that displays a TabsComponent. Once mounted, it connects to the back-end to obtain data
 * for the from date, to date, location, and weather variable selected.
 * 
 * @component
 */
function ChartPopup(props) {

  const [data, setData] = useState(undefined);
  const [zoomDomain, setZoomDomain] = useState(undefined);
  const [minMax, setMinMax] = useState(false);

  useEffect(() => {

    let active = true;

    async function obtainData() {

      const uri = expandLink(props.linksProperty, props.location, props.curVar, props.fromDate, props.toDate);

      const response = await fetch(uri, {
        method: 'GET',
        headers: new Headers({
          'Content-Type': 'application/json'
        })
      });
      const dataArray = await response.json();

      if (active) {
        const n = dataArray.length;

        for (let i = 0; i < n; i++) {
          const dateS = dataArray[i].date;
          dataArray[i].date = new Date(dateS.substring(0, DLEN));

        }

        if (n > 0) {

          const fd = dataArray[0].date;
          const td = dataArray[dataArray.length - 1].date;

          if (props.curVar === PRECIP && dataArray[0].precipitation.minvalue !== undefined) {

            setMinMax(true);
          } else {
            setMinMax(false);
          }

          setData(dataArray);
          setZoomDomain({ x: [fd, td] });
        } else {

          setData(undefined);
          setZoomDomain(undefined);
          setMinMax(false);
        }
      }

    }

    obtainData();
    const cancel = () => active = false;
    return cancel;

  }, [props.linksProperty, props.curVar, props.location, props.fromDate, props.toDate]);

  function handleClick(e) {
    e.stopPropagation();
  }

  const vc = chartText(props.curVar);

  return (
    <div id="popupdiv" data-testid='chart-div' onClick={handleClick}>
      <div id="iicon">
        <IoMdClose onClick={props.closePopup} size={20} color="rgb(192, 57, 43)" />

      </div>
      <center>
        {vc} data from {props.location.name.trim()}

        {data && <TabsComponent curVar={props.curVar} data={data} zoomDomain={zoomDomain} minMax={minMax}
          setZoomDomain={setZoomDomain} />}

      </center>
    </div>
  );


}

ChartPopup.propTypes = {
  /** Specifies the weather variable currently selected. */
  curVar: PropTypes.string.isRequired,

  /** Specifies the properties of the location selected. */
  location: PropTypes.object.isRequired,

  /** Specifies the start date for the data that is to be displayed. */
  fromDate: PropTypes.string.isRequired,

  /** Specifies the end date for the data that is to be displayed. */
  toDate: PropTypes.string.isRequired,

  /** Called when the close icon is clicked. */
  closePopup: PropTypes.func.isRequired,

  linksProperty: PropTypes.object.isRequired,
}

export default ChartPopup;

