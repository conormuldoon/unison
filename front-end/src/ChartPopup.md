ChartPopup example:

```js
import {useState} from 'react';
const [display,setDisplay] = useState(false);
function toggleDisplay(){
    if(display)setDisplay(false);
    else setDisplay(true);
}

if(display){
    window.scrollTo(0,0);
}

<div>
    
    <button onClick={toggleDisplay}>{(display)?"Hide":"Display"}</button>
    {display&&<ChartPopup closePopup={toggleDisplay} varCur="Precipitation" location="UCD" fromDate="4/10/2019" toDate="5/10/2019" />}
</div>
```