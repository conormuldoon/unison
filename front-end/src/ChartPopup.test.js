import fetchMock from 'fetch-mock';
import React from 'react';
import ReactDOM from 'react-dom';
import { render } from "react-testing-library";
import ChartPopup from './ChartPopup';


const fromDate = '1/2/2018';
const toDate = '7/10/2019';
const chartPopup = <ChartPopup varCur='Temperature' location='UCD'
  fromDate={fromDate} toDate={toDate} closePopup={() => { }} />;

const apiRequest = 'end:/temperature?location=UCD&fromDate=' + fromDate + '&toDate=' + toDate;

it('renders without crashing', async () => {


  fetchMock.get(apiRequest, []);

  const div = document.createElement('div');

  ReactDOM.render(chartPopup, div);
  ReactDOM.unmountComponentAtNode(div);
  fetchMock.restore();
});



it('mathes snapshot', () => {

  fetchMock.get(apiRequest, []);

  const { container } = render(chartPopup);

  expect(container).toMatchSnapshot();

  fetchMock.restore();

});

it('display lower case for second word in variable', () => {

  fetchMock.get('end:/cloudLevel?location=UCD&fromDate=' + fromDate + '&toDate=' + toDate, []);

  const { getByText }=render(<ChartPopup varCur='Cloud Level' location='UCD' fromDate={fromDate} toDate={toDate} closePopup={() => { }} />);
  getByText('Cloud level data from UCD');
  
  fetchMock.restore();
});
