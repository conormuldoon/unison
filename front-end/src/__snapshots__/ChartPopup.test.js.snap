// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes snapshot 1`] = `
<div>
  <div
    data-testid="chart-div"
    id="popupdiv"
  >
    <div
      id="iicon"
    >
      <svg
        color="rgb(192, 57, 43)"
        fill="currentColor"
        height="20"
        size="20"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(192, 57, 43);"
        viewBox="0 0 512 512"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
        />
      </svg>
      <button
        aria-label="email"
        class="react-share__ShareButton"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          height="24"
          viewBox="0 0 64 64"
          width="24"
        >
          <rect
            fill="#7f7f7f"
            height="64"
            rx="0"
            ry="0"
            width="64"
          />
          <path
            d="M17,22v20h30V22H17z M41.1,25L32,32.1L22.9,25H41.1z M20,39V26.6l12,9.3l12-9.3V39H20z"
            fill="white"
          />
        </svg>
      </button>
      <button
        aria-label="whatsapp"
        class="react-share__ShareButton"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          height="24"
          viewBox="0 0 64 64"
          width="24"
        >
          <rect
            fill="#25D366"
            height="64"
            rx="0"
            ry="0"
            width="64"
          />
          <path
            d="m42.32286,33.93287c-0.5178,-0.2589 -3.04726,-1.49644 -3.52105,-1.66732c-0.4712,-0.17346 -0.81554,-0.2589 -1.15987,0.2589c-0.34175,0.51004 -1.33075,1.66474 -1.63108,2.00648c-0.30032,0.33658 -0.60064,0.36247 -1.11327,0.12945c-0.5178,-0.2589 -2.17994,-0.80259 -4.14759,-2.56312c-1.53269,-1.37217 -2.56312,-3.05503 -2.86603,-3.57283c-0.30033,-0.5178 -0.03366,-0.80259 0.22524,-1.06149c0.23301,-0.23301 0.5178,-0.59547 0.7767,-0.90616c0.25372,-0.31068 0.33657,-0.5178 0.51262,-0.85437c0.17088,-0.36246 0.08544,-0.64725 -0.04402,-0.90615c-0.12945,-0.2589 -1.15987,-2.79613 -1.58964,-3.80584c-0.41424,-1.00971 -0.84142,-0.88027 -1.15987,-0.88027c-0.29773,-0.02588 -0.64208,-0.02588 -0.98382,-0.02588c-0.34693,0 -0.90616,0.12945 -1.37736,0.62136c-0.4712,0.5178 -1.80194,1.76053 -1.80194,4.27186c0,2.51134 1.84596,4.945 2.10227,5.30747c0.2589,0.33657 3.63497,5.51458 8.80262,7.74113c1.23237,0.5178 2.1903,0.82848 2.94111,1.08738c1.23237,0.38836 2.35599,0.33657 3.24402,0.20712c0.99159,-0.15534 3.04985,-1.24272 3.47963,-2.45956c0.44013,-1.21683 0.44013,-2.22654 0.31068,-2.45955c-0.12945,-0.23301 -0.46601,-0.36247 -0.98382,-0.59548m-9.40068,12.84407l-0.02589,0c-3.05503,0 -6.08417,-0.82849 -8.72495,-2.38189l-0.62136,-0.37023l-6.47252,1.68286l1.73463,-6.29129l-0.41424,-0.64725c-1.70875,-2.71846 -2.6149,-5.85116 -2.6149,-9.07706c0,-9.39809 7.68934,-17.06155 17.15993,-17.06155c4.58253,0 8.88029,1.78642 12.11655,5.02268c3.23625,3.21036 5.02267,7.50812 5.02267,12.06476c-0.0078,9.3981 -7.69712,17.06155 -17.14699,17.06155m14.58906,-31.58846c-3.93529,-3.80584 -9.1133,-5.95471 -14.62789,-5.95471c-11.36055,0 -20.60848,9.2065 -20.61625,20.52564c0,3.61684 0.94757,7.14565 2.75211,10.26282l-2.92557,10.63564l10.93337,-2.85309c3.0136,1.63108 6.4052,2.4958 9.85634,2.49839l0.01037,0c11.36574,0 20.61884,-9.2091 20.62403,-20.53082c0,-5.48093 -2.14111,-10.64081 -6.03239,-14.51915"
            fill="white"
          />
        </svg>
      </button>
      <button
        aria-label="telegram"
        class="react-share__ShareButton"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          height="24"
          viewBox="0 0 64 64"
          width="24"
        >
          <rect
            fill="#37aee2"
            height="64"
            rx="0"
            ry="0"
            width="64"
          />
          <path
            d="m45.90873,15.44335c-0.6901,-0.0281 -1.37668,0.14048 -1.96142,0.41265c-0.84989,0.32661 -8.63939,3.33986 -16.5237,6.39174c-3.9685,1.53296 -7.93349,3.06593 -10.98537,4.24067c-3.05012,1.1765 -5.34694,2.05098 -5.4681,2.09312c-0.80775,0.28096 -1.89996,0.63566 -2.82712,1.72788c-0.23354,0.27218 -0.46884,0.62161 -0.58825,1.10275c-0.11941,0.48114 -0.06673,1.09222 0.16682,1.5716c0.46533,0.96052 1.25376,1.35737 2.18443,1.71383c3.09051,0.99037 6.28638,1.93508 8.93263,2.8236c0.97632,3.44171 1.91401,6.89571 2.84116,10.34268c0.30554,0.69185 0.97105,0.94823 1.65764,0.95525l-0.00351,0.03512c0,0 0.53908,0.05268 1.06412,-0.07375c0.52679,-0.12292 1.18879,-0.42846 1.79109,-0.99212c0.662,-0.62161 2.45836,-2.38812 3.47683,-3.38552l7.6736,5.66477l0.06146,0.03512c0,0 0.84989,0.59703 2.09312,0.68132c0.62161,0.04214 1.4399,-0.07726 2.14229,-0.59176c0.70766,-0.51626 1.1765,-1.34683 1.396,-2.29506c0.65673,-2.86224 5.00979,-23.57745 5.75257,-27.00686l-0.02107,0.08077c0.51977,-1.93157 0.32837,-3.70159 -0.87096,-4.74991c-0.60054,-0.52152 -1.2924,-0.7498 -1.98425,-0.77965l0,0.00176zm-0.2072,3.29069c0.04741,0.0439 0.0439,0.0439 0.00351,0.04741c-0.01229,-0.00351 0.14048,0.2072 -0.15804,1.32576l-0.01229,0.04214l-0.00878,0.03863c-0.75858,3.50668 -5.15554,24.40802 -5.74203,26.96472c-0.08077,0.34417 -0.11414,0.31959 -0.09482,0.29852c-0.1756,-0.02634 -0.50045,-0.16506 -0.52679,-0.1756l-13.13468,-9.70175c4.4988,-4.33199 9.09945,-8.25307 13.744,-12.43229c0.8218,-0.41265 0.68483,-1.68573 -0.29852,-1.70681c-1.04305,0.24584 -1.92279,0.99564 -2.8798,1.47502c-5.49971,3.2626 -11.11882,6.13186 -16.55882,9.49279c-2.792,-0.97105 -5.57873,-1.77704 -8.15298,-2.57601c2.2336,-0.89555 4.00889,-1.55579 5.75608,-2.23009c3.05188,-1.1765 7.01687,-2.7042 10.98537,-4.24067c7.94051,-3.06944 15.92667,-6.16346 16.62028,-6.43037l0.05619,-0.02283l0.05268,-0.02283c0.19316,-0.0878 0.30378,-0.09658 0.35471,-0.10009c0,0 -0.01756,-0.05795 -0.00351,-0.04566l-0.00176,0zm-20.91715,22.0638l2.16687,1.60145c-0.93418,0.91311 -1.81743,1.77353 -2.45485,2.38812l0.28798,-3.98957"
            fill="white"
          />
        </svg>
      </button>
    </div>
    <center>
      <div
        style="margin-top: 10px; margin-bottom: 20px;"
      >
        Precipitation data from UCD
      </div>
      <div
        id="chartdiv"
      >
        <div
          class="react-tabs"
          data-tabs="true"
          id="tabscomponent"
        >
          <ul
            class="react-tabs__tab-list"
            role="tablist"
          >
            <li
              aria-controls="react-tabs-1"
              aria-disabled="false"
              aria-selected="false"
              class="react-tabs__tab"
              id="react-tabs-0"
              role="tab"
            >
              20ᵗʰ percentile
            </li>
            <li
              aria-controls="react-tabs-3"
              aria-disabled="false"
              aria-selected="true"
              class="react-tabs__tab react-tabs__tab--selected"
              id="react-tabs-2"
              role="tab"
              tabindex="0"
            >
              Median
            </li>
            <li
              aria-controls="react-tabs-5"
              aria-disabled="false"
              aria-selected="false"
              class="react-tabs__tab"
              id="react-tabs-4"
              role="tab"
            >
              80ᵗʰ percentile
            </li>
          </ul>
          <div
            aria-labelledby="react-tabs-0"
            class="react-tabs__tab-panel"
            id="react-tabs-1"
            role="tabpanel"
          />
          <div
            aria-labelledby="react-tabs-2"
            class="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-3"
            role="tabpanel"
          >
            <div
              data-testid="chart"
              id="tabdiv"
            >
              <div
                class="VictoryContainer"
                style="max-width: 55%; max-height: 55%; height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
              >
                <svg
                  height="350"
                  role="img"
                  style="pointer-events: all; width: 100%; height: auto;"
                  viewBox="0 0 350 350"
                  width="350"
                >
                  <g
                    clip-path="url(#victory-clip-2)"
                    style="border: 1px solid #ccc; width: 100%; height: 100%;"
                  >
                    <defs>
                      <clippath
                        id="victory-clip-2"
                      >
                        <rect
                          height="290"
                          vector-effect="non-scaling-stroke"
                          width="280"
                          x="70"
                          y="10"
                        />
                      </clippath>
                    </defs>
                    <path
                      d="M70,300L73.6842105263158,300L77.36842105263159,300L81.05263157894736,300L84.73684210526316,300L88.42105263157895,300L92.10526315789474,300L95.78947368421052,300L99.47368421052632,300L103.15789473684211,300L106.84210526315789,300L110.52631578947368,300L114.21052631578948,300L117.89473684210526,300L121.57894736842104,300L125.26315789473685,300L128.94736842105263,300L132.6315789473684,300L136.31578947368422,300L140,300L143.68421052631578,300L147.3684210526316,300L151.05263157894737,300L154.73684210526315,300L158.42105263157896,300L162.10526315789474,300L165.78947368421052,300L169.4736842105263,300L173.1578947368421,300L176.8421052631579,300L180.5263157894737,300L184.21052631578948,300L187.89473684210526,300L191.57894736842107,300L195.26315789473682,300L198.94736842105263,300L202.6315789473684,300L206.31578947368422,300L210,300L213.68421052631578,300L217.36842105263156,300L221.05263157894737,300L224.73684210526318,300L228.42105263157893,300L232.10526315789474,300L235.78947368421052,300L239.4736842105263,300L243.1578947368421,300L246.8421052631579,300L250.5263157894737,300L254.21052631578948,300L257.89473684210526,300L261.57894736842104,300L265.2631578947368,300L268.9473684210526,300L272.63157894736844,300L276.3157894736842,300L280,300L283.6842105263158,300L287.36842105263156,300L291.0526315789474,300L294.7368421052632,300L298.42105263157896,300L302.10526315789474,300L305.7894736842105,300L309.47368421052636,300L313.15789473684214,300L316.84210526315786,300L320.52631578947364,300L324.2105263157895,300L327.89473684210526,300L331.57894736842104,300L335.2631578947368,300L338.9473684210526,300L342.63157894736844,300L346.3157894736842,300L350,300L353.6842105263158,300L357.36842105263156,300L361.0526315789474,300L364.7368421052631,300L368.42105263157896,300L372.10526315789474,300L375.7894736842105,300L379.47368421052636,300L383.15789473684214,300L386.84210526315786,300L390.52631578947364,300L394.2105263157895,300L397.89473684210526,300L401.57894736842104,300L405.2631578947369,300L408.9473684210526,300L412.63157894736844,300L416.3157894736842,300L420,300L423.6842105263158,300L427.36842105263156,300L431.0526315789474,300L434.7368421052631,300L438.42105263157896,300L442.10526315789474,300L445.7894736842105,300L449.47368421052636,300L453.15789473684214,300L456.84210526315786,300L460.52631578947364,10"
                      role="presentation"
                      shape-rendering="auto"
                      style="fill: transparent; stroke: #c43a31; opacity: 1; stroke-width: 2;"
                    />
                  </g>
                  <g
                    role="presentation"
                  >
                    <line
                      role="presentation"
                      shape-rendering="auto"
                      style="stroke: #90A4AE; fill: transparent; stroke-width: 2; stroke-linecap: round; stroke-linejoin: round;"
                      vector-effect="non-scaling-stroke"
                      x1="70"
                      x2="350"
                      y1="300"
                      y2="300"
                    />
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="88.42105263157895"
                        x2="88.42105263157895"
                        y1="300"
                        y2="10"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="88.42105263157895"
                        x2="88.42105263157895"
                        y1="300"
                        y2="305"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="10.26"
                        id="chart-axis-1-tickLabels-0"
                        x="88.42105263157895"
                        y="313"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="middle"
                          x="88.42105263157895"
                        >
                          Feb 24
                        </tspan>
                      </text>
                    </g>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="132.6315789473684"
                        x2="132.6315789473684"
                        y1="300"
                        y2="10"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="132.6315789473684"
                        x2="132.6315789473684"
                        y1="300"
                        y2="305"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="10.26"
                        id="chart-axis-1-tickLabels-1"
                        x="132.6315789473684"
                        y="313"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="middle"
                          x="132.6315789473684"
                        >
                          12 PM
                        </tspan>
                      </text>
                    </g>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="176.8421052631579"
                        x2="176.8421052631579"
                        y1="300"
                        y2="10"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="176.8421052631579"
                        x2="176.8421052631579"
                        y1="300"
                        y2="305"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="10.26"
                        id="chart-axis-1-tickLabels-2"
                        x="176.8421052631579"
                        y="313"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="middle"
                          x="176.8421052631579"
                        >
                          Mon 25
                        </tspan>
                      </text>
                    </g>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="221.05263157894737"
                        x2="221.05263157894737"
                        y1="300"
                        y2="10"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="221.05263157894737"
                        x2="221.05263157894737"
                        y1="300"
                        y2="305"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="10.26"
                        id="chart-axis-1-tickLabels-3"
                        x="221.05263157894737"
                        y="313"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="middle"
                          x="221.05263157894737"
                        >
                          12 PM
                        </tspan>
                      </text>
                    </g>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="265.2631578947368"
                        x2="265.2631578947368"
                        y1="300"
                        y2="10"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="265.2631578947368"
                        x2="265.2631578947368"
                        y1="300"
                        y2="305"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="10.26"
                        id="chart-axis-1-tickLabels-4"
                        x="265.2631578947368"
                        y="313"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="middle"
                          x="265.2631578947368"
                        >
                          Tue 26
                        </tspan>
                      </text>
                    </g>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="309.47368421052636"
                        x2="309.47368421052636"
                        y1="300"
                        y2="10"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="309.47368421052636"
                        x2="309.47368421052636"
                        y1="300"
                        y2="305"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="10.26"
                        id="chart-axis-1-tickLabels-5"
                        x="309.47368421052636"
                        y="313"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="middle"
                          x="309.47368421052636"
                        >
                          12 PM
                        </tspan>
                      </text>
                    </g>
                  </g>
                  <g
                    role="presentation"
                  >
                    <line
                      role="presentation"
                      shape-rendering="auto"
                      style="stroke: #90A4AE; fill: transparent; stroke-width: 2; stroke-linecap: round; stroke-linejoin: round;"
                      vector-effect="non-scaling-stroke"
                      x1="70"
                      x2="70"
                      y1="10"
                      y2="300"
                    />
                    <text
                      direction="inherit"
                      dx="0"
                      dy="-1.7400000000000002"
                      id="chart-axis-2-axisLabel-0"
                      transform="rotate(-90,25,155)"
                      x="25"
                      y="155"
                    >
                      <tspan
                        dx="0"
                        style="padding: 45px; text-anchor: middle; font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; fill: #455A64; stroke: transparent; stroke-width: 0;"
                        text-anchor="middle"
                        x="25"
                      >
                        Millimetres
                      </tspan>
                    </text>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="70"
                        x2="350"
                        y1="227.5"
                        y2="227.5"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="70"
                        x2="65"
                        y1="227.5"
                        y2="227.5"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="4.26"
                        id="chart-axis-2-tickLabels-0"
                        x="57"
                        y="227.5"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="end"
                          x="57"
                        >
                          0.05
                        </tspan>
                      </text>
                    </g>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="70"
                        x2="350"
                        y1="155"
                        y2="155"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="70"
                        x2="65"
                        y1="155"
                        y2="155"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="4.26"
                        id="chart-axis-2-tickLabels-1"
                        x="57"
                        y="155"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="end"
                          x="57"
                        >
                          0.1
                        </tspan>
                      </text>
                    </g>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="70"
                        x2="350"
                        y1="82.50000000000003"
                        y2="82.50000000000003"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="70"
                        x2="65"
                        y1="82.50000000000003"
                        y2="82.50000000000003"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="4.26"
                        id="chart-axis-2-tickLabels-2"
                        x="57"
                        y="82.50000000000003"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="end"
                          x="57"
                        >
                          0.15
                        </tspan>
                      </text>
                    </g>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                        vector-effect="non-scaling-stroke"
                        x1="70"
                        x2="350"
                        y1="10"
                        y2="10"
                      />
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="70"
                        x2="65"
                        y1="10"
                        y2="10"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        dy="4.26"
                        id="chart-axis-2-tickLabels-3"
                        x="57"
                        y="10"
                      >
                        <tspan
                          dx="0"
                          style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                          text-anchor="end"
                          x="57"
                        >
                          0.2
                        </tspan>
                      </text>
                    </g>
                  </g>
                </svg>
                <div
                  style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: auto;"
                >
                  <svg
                    height="350"
                    style="overflow: visible; width: 100%; height: auto;"
                    viewBox="0 0 350 350"
                    width="350"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="react-tabs-4"
            class="react-tabs__tab-panel"
            id="react-tabs-5"
            role="tabpanel"
          />
        </div>
      </div>
    </center>
  </div>
</div>
`;
